frontend/src/hooks/useAuthRedirect.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router';
+import { useAuthStore } from '@/store/authStore';
+
+const getAccessTokenFromCookie = () => {
+  const match = document.cookie.match(/(?:^|; )accessToken=([^;]*)/);
+  return match ? decodeURIComponent(match[1]) : null;
+};
+
+/**
+ * 인증 상태를 검사하고, 토큰이 있으면 /matrix로,
+ * 없으면 /login으로 이동
+ */
+export const useAuthRedirect = () => {
+  const navigate = useNavigate();
+  const { token, setToken } = useAuthStore();
+
+  useEffect(() => {
+    if (!token) {
+      const cookieToken = getAccessTokenFromCookie();
+      if (cookieToken) {
+        setToken(cookieToken);
+        navigate('/matrix');
+        return;
+      }
+    }
+
+    if (token) {
+      navigate('/matrix');
+    } else {
+      navigate('/login');
+    }
+  }, [token, setToken, navigate]);
```

Comment:
처음에 코드 봤을 때 조건문의 흐름이 한 눈에 들어오지 않았어서, 어떻게 개선할 수 있을까 고민해봤습니다! 
각 조건마다 **early return 패턴**을 적용하면, 가능한 모든 상태 (토큰이 스토어에 있는 경우, 쿠키에만 있는 경우, 아예 없는 경우)를 더 가독성이 좋게 처리할 수 있을 것 같아서 제안드려 봅니다! 

```suggestion
   useEffect(() => {
    // 스토어에 토큰이 있는 경우
    if (token) {
      navigate('/matrix');
      return;
    }
    
    // 스토어에 토큰이 없지만 쿠키에 토큰이 있는 경우
    const cookieToken = getAccessTokenFromCookie();
    if (cookieToken) {
      setToken(cookieToken);
      navigate('/matrix');
      return;
    }
    
    // 토큰이 어디에도 없는 경우
    navigate('/login');
  }, [token, setToken, navigate]);
```

Reply:
너무 좋은 것 같아요!!! 반영하겠습니다 좋은 지식 알아갑니다!! 🥹